src/vs/workbench/contrib/terminal/browser/xterm/quickFixAddon.ts

```diff
@@ -143,8 +159,14 @@ export class TerminalQuickFixAddon extends Disposable implements ITerminalAddon,
 		this._quickFixes = fixes;
 		this._register(onDidRunQuickFix((id) => {
 			const info = `terminal/quick-fix/${id}`;
-			this._logService.debug(info);
-			this._telemetryService?.publicLog2<{ classification: 'SystemMetaData'; purpose: 'FeatureInsight' }>(info);
+			this._logService.debug(info, {
+				fixesShown: this._fixesShown,
+				expectedCommand: this._expectedCommands?.includes(command.command) || false,
+			});
+			this._telemetryService?.publicLog2<QuickFixResultTelemetryEvent, QuickFixClassification>(info, {
```

Comment:
This will introduce many new events that would all need to be documented, instead the `quickFixId` should be done similar to `providerId` here:

https://github.com/microsoft/vscode/blob/db4c864178ea96ae85480c9a8996b033db372aa5/src/vs/editor/contrib/suggest/browser/suggestController.ts#L463-L464